plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/IngestModule.java

```diff
@@ -53,23 +55,23 @@ protected void configure() {
         binder().bind(PipelineStore.class).asEagerSingleton();
         binder().bind(SimulateExecutionService.class).asEagerSingleton();
 
-        addProcessor(GeoIpProcessor.TYPE, new GeoIpProcessor.Factory());
-        addProcessor(GrokProcessor.TYPE, new GrokProcessor.Factory());
-        addProcessor(DateProcessor.TYPE, new DateProcessor.Factory());
-        addProcessor(SetProcessor.TYPE, new SetProcessor.Factory());
-        addProcessor(RenameProcessor.TYPE, new RenameProcessor.Factory());
-        addProcessor(RemoveProcessor.TYPE, new RemoveProcessor.Factory());
-        addProcessor(SplitProcessor.TYPE, new SplitProcessor.Factory());
-        addProcessor(JoinProcessor.TYPE, new JoinProcessor.Factory());
-        addProcessor(UppercaseProcessor.TYPE, new UppercaseProcessor.Factory());
-        addProcessor(LowercaseProcessor.TYPE, new LowercaseProcessor.Factory());
-        addProcessor(TrimProcessor.TYPE, new TrimProcessor.Factory());
-        addProcessor(ConvertProcessor.TYPE, new ConvertProcessor.Factory());
-        addProcessor(GsubProcessor.TYPE, new GsubProcessor.Factory());
-        addProcessor(MetaDataProcessor.TYPE, new MetaDataProcessor.Factory());
+        addProcessor(GeoIpProcessor.TYPE, environment -> new GeoIpProcessor.Factory(environment.configFile()));
```

Comment:
I honestly don't get it... the way I look at it, we moved from having the `env` "polluting" the `Factory` interface to having it now "pollute" the `Factory.Provider`... why don't we just either wire the concrete factories into a map binder, or, if we don't want to wire them, see if we can pass in the Env. directly to the Module ctor (just like it can accept `Settings`)